tests/unit/database_tests.py

```diff
@@ -202,6 +202,11 @@ def test_create_db_delete_db(self):
             # No issue should arise if attempting to create existing database
             db_2 = db.create()
             self.assertEqual(db, db_2)
+            # If we use throw_on_exists=True, it will arise a
+            # CloudantDatabaseException if the database already exists.
+            with self.assertRaises(CloudantDatabaseException) as cm:
+                db.create(throw_on_exists=True)
+            self.assertEqual(cm.exception.status_code, 100)
```

Comment:
The `CloudantDatabaseException` should basically take the original response status code and use it as its own code.  The goal here was not to mask the actual error code.  So my assumption here was that a 409 would be returned by the `raise_for_status` here: https://github.com/porthunt/python-cloudant/blob/f823d1f68e3431ee63f9f6f4cd9d93a0bf3e39e0/src/cloudant/database.py#L351-L353.  That 409 would then be re-raised as a `CloudantDatabaseException` preserving the original HTTP response code and then re-raised again later as a `CloudantClientException` while still preserving the original HTTP response code.  It appears that the 409 assumption was incorrect?  If so then we have a different problem to deal with.